pkgs/development/libraries/nng/default.nix

```diff
@@ -11,15 +11,21 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-qbjMLpPk5FxH710Mf8AIraY0mERbaxVVhTT94W0EV+k=";
   };
 
-  nativeBuildInputs = [ cmake ninja ];
+  nativeBuildInputs = [ cmake ninja ]
+    ++ lib.optionals mbedtlsSupport [ mbedtls ];
 
-  cmakeFlags = [ "-G Ninja" ];
+  buildInputs = lib.optional mbedtlsSupport [ mbedtls ];
+
+  cmakeFlags = [ "-G Ninja" ]
+    ++ lib.optionals mbedtlsSupport [ "-DMBEDTLS_ROOT_DIR=${mbedtls}" ]
+    ++ lib.optionals (tlsSupport || mbedtlsSupport) [ "-DNNG_ENABLE_TLS=ON" ];
```

Comment:
I don't think we need the tlsSupport flag. It does not reduce dependencies and having tls support is something you'd always want I guess.

Reply:
Done